src/PhpSpreadsheet/Style/NumberFormat/Formatter.php

```diff
@@ -126,7 +126,10 @@ public static function toFormattedString($value, string $format, ?array $callBac
         // For now we do not treat strings in sections, although section 4 of a format code affects strings
         // Process a single block format code containing @ for text substitution
         if (preg_match(self::SECTION_SPLIT, $format) === 0 && preg_match(self::SYMBOL_AT, $format) === 1) {
-            return str_replace('"', '', preg_replace(self::SYMBOL_AT, (string) $value, $format) ?? '');
+            //escape any dollar signs on the string, so they are not replaced with an empty value
+            $value = str_replace('$', '\\$', (string) $value);
+
+            return str_replace('"', '', preg_replace(self::SYMBOL_AT, $value, $format) ?? '');
```

Comment:
I know that you did not introduce this statement. However, since you've brought up this can of worms, does this statement cause unexpected results when the cell contains quote marks? If so, might you be able to address that problem as well?

Reply:
I never faced this issue, but I can check.

Reply:
@oleibman There are unexpected results when the cell contains quote marks, but I'm not sure how to fix it. Because I don't know why that str_replace was put in the first place.

Do you have information about it? I will check the commit history to see if I find anything about it.

Reply:
I think it has been there for a long time. I'm not at all sure that the `str_replace` is needed there. I will do more research. In the meantime, what happens if you just remove it (so `return preg_replace(...) ?? '`)? Does it fix the problem I asked about, and does it break any existing tests?

Reply:
Thanks for replying. I finally got it solved and just pushed the fix. We just ignore the text formatting when the format is a single `@`.

It made the overall things simpler.

I think there might be still one issue, if a cell has custom formatting and that custom formatting contains a quotation mark, it would get erased. But I couldn't create this example, neither in Excel or Google Sheets.

Reply:
I apologize, I don't mean to be giving you a runaround here. You pushed a change while I was adding another comment. And we were certainly on the same track (adding a check to see if the the format string consisted entirely of `@`). At this point, I think it still needs a bit more work, which I will discuss below in your test member.

Reply:
No problem, I'm glad to help o/

I will mark this thread as resolved